src/Input.js

```diff
@@ -245,7 +245,12 @@ const Input = React.createClass({
       ];
     }
 
-    return <FormGroup {...this.props}>{children}</FormGroup>;
+    if (this.props.type === 'submit') {
+      let {bsStyle, ...other} = this.props; /* eslint no-unused-vars: 0 */
```

Comment:
I've pondered about it.
The problem with that particular case is that only developer knows for sure is the var necessary or not. So it has to be explicitly marked.